pkg/config/parsing.go

```diff
@@ -87,82 +87,109 @@ func loadYaml(inputFile string, conf *SwoopConfig) error {
 }
 
 func ValidateConfig(sc *SwoopConfig) error {
-
-	for wf_name, wf := range sc.Workflows {
-		for cb_name, cb := range wf.Callbacks {
+	for wfName, wf := range sc.Workflows {
+		for cbName, cb := range wf.Callbacks {
+			var (
+				callbackParamNames []string
+				handlerParamNames  []string
+				requiredParamNames []string
+			)
 
 			// Check for conflicts in on/not on conditions
-
 			on := cb.On
 			notOn := cb.NotOn
 			for _, v := range on {
 				if utils.Contains(notOn, v) {
-					err := fmt.Errorf("the workflow status value '%v' for callback '%s' in workflow '%s' is conflicting and appears in both the on/ notOn clauses in the config", v, cb_name, wf_name)
-					fmt.Println(err)
+					return fmt.Errorf(
+						"the workflow status value '%v' for callback '%s' in workflow '%s' is conflicting and appears in both the on/ notOn clauses in the config",
+						v,
+						cbName,
+						wfName,
+					)
 				}
 			}
 
-			callback_params := cb.Parameters
-			handler_name := cb.Handler
-			handler_params := sc.Handlers[handler_name].Parameters
-			var callback_param_names []string
-			var handler_param_names []string
+			callbackParams := cb.Parameters
+			handlerName := cb.Handler
 
-			for hand_name, hand_param := range handler_params {
-				if hand_param.IsRequired() {
-					handler_param_names = append(handler_param_names, hand_name)
+			for name, param := range sc.Handlers[handlerName].Parameters {
+				handlerParamNames = append(handlerParamNames, name)
+				if param.IsRequired() {
+					requiredParamNames = append(requiredParamNames, name)
 				}
 			}
 
-			for cb_name := range callback_params {
-				callback_param_names = append(callback_param_names, cb_name)
+			for name := range callbackParams {
+				callbackParamNames = append(callbackParamNames, name)
 			}
 
 			// Check if all required handler parameters are defined in callback
-			for _, v := range handler_param_names {
-				if !utils.Contains(callback_param_names, v) {
-					return fmt.Errorf("required handler parameter '%s' is not provided as a callback parameter for callback '%s' in workflow '%s' ", v, cb_name, wf_name)
+			for _, v := range handlerParamNames {
```

Comment:
is this checking just the required handler parameters, or all handler parameters? I thought we wanted to check only the required handler parameters.

Reply:
I believe I screwed this up while refactoring. I'll push a fix.